client/src/loaders/PromiseLoader.js

```diff
@@ -24,18 +24,21 @@ class PromiseLoader extends React.Component {
   componentDidMount() {
     const {promiseFn} = this.props;
     promiseFn()
-      .then(this.onResolved)
-      .then(this.onRejected);
+      .then(result => {
```

Comment:
I think it's best if we let `result` stay a black box, so that it can be whatever people using this component want.  The state of `promiseFn()` should be the only thing that determines what state it's in (eg resolve/reject/pending).

With the changes to `JsonLoader` over [here](https://github.com/mit-teaching-systems-lab/swipe-right-for-cs/pull/93#discussion_r162464650), would it work to remove this if check?  We're essentially moving this check to be `JsonLoader`'s job (since it knows that the response should be valid json) rather than `PromiseLoader`'s job, so it can stay generic and just have to track the pending/resolve/reject status of the promise.

Reply:
I think I get what you're saying. Just changed this and did some basic manual tests to make sure it works.